src/snowflake/snowpark/_internal/compiler/plan_compiler.py

```diff
@@ -151,8 +154,7 @@ def compile(self) -> Dict[PlanQueryType, List[Query]]:
             return queries
         else:
             final_plan = self._plan
-            if self._plan.session.cte_optimization_enabled:
-                final_plan = final_plan.replace_repeated_subquery_with_cte()
+            final_plan = final_plan.replace_repeated_subquery_with_cte()
```

Comment:
Is cte on by default now?

Reply:
This check is already done inside `replace_repeated_subquery_with_cte`. the check here was redundant.